CenterForOpenScience/osf.io
# [PLAT-920][Hotfix] Implement user deletion protocol

## Purpose

We need a GDPR compliant delete method, this gives us one.

## Changes

- add method gdpr_delete to user model
- adds management command wrapper for method
- a whole bunch of tests.

## QA Notes

Use the management command `python manage.py gdpr_delete_user guid1` to delete different users and see that all their identifying info is striped and that there are no references to them on the site. If you attempt to delete a user with preprints or registrations or a user who is an admin on a shared project, the delete should fail.

## Documentation

This feature is for internal use only.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/PLAT-920
